terraform/modules/tf_stream_alert_flow_logs/iam.tf

```diff
@@ -1,98 +1,126 @@
-    {
-      "Action": [
-        "logs:CreateLogGroup",
-        "logs:CreateLogStream",
-        "logs:PutLogEvents",
-        "logs:DescribeLogGroups",
-        "logs:DescribeLogStreams"
-      ],
-      "Effect": "Allow",
-      "Resource": "*"
-    }
-  ]
-}   
-EOF
+  policy = "${data.aws_iam_policy_document.flow_log_put_cloudwatch_logs.json}"
+}
+
+// IAM Policy Doc: CloudWatch Put Events
+data "aws_iam_policy_document" "flow_log_put_cloudwatch_logs" {
+  statement {
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:DescribeLogGroups",
+      "logs:DescribeLogStreams",
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
 }
 
 // Allow log subscription to write to Kinesis Stream
 // http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/SubscriptionFilters.html
-
+// IAM Role: Clustered CloudWatch Flow Log Role
 resource "aws_iam_role" "flow_log_subscription_role" {
-  name = "flow_log_subscription_role"
-
-  assume_role_policy = <<EOF
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Sid": "",
-      "Effect": "Allow",
-      "Principal": {
-        "Service": "logs.${var.region}.amazonaws.com"
-      },
-      "Action": "sts:AssumeRole"
+  name = "stream_alert_${var.cluster}_flow_log_subscription_role"
+
+  assume_role_policy = "${data.aws_iam_policy_document.cloudwatch_logs_assume_role_policy.json}"
+}
+
+// IAM Policy Doc: AssumeRole for CloudWatch Logs
+data "aws_iam_policy_document" "cloudwatch_logs_assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["logs.${var.region}.amazonaws.com"]
     }
-  ]
-} 
-EOF
+  }
 }
 
+// IAM Policy: Write to Kinesis
 resource "aws_iam_role_policy" "flow_logs_kinesis_wo" {
   name = "write_flow_logs_to_kinesis"
   role = "${aws_iam_role.flow_log_subscription_role.id}"
 
-  policy = <<EOF
-{
-  "Version": "2012-10-17",
-  "Statement" : [
-    {
-      "Action": [
-        "kinesis:PutRecord*",
-        "kinesis:DescribeStream",
-        "kinesis:ListStreams"
-      ],
-      "Effect": "Allow",
-      "Resource": [
-        "${var.destination_stream_arn}"
+  policy = "${data.aws_iam_policy_document.flow_logs_put_kinesis_events.json}"
+}
+
+// IAM Policy Doc: Write to Kinesis
+data "aws_iam_policy_document" "flow_logs_put_kinesis_events" {
+  statement {
+    actions = [
+      "kinesis:PutRecord*",
+      "kinesis:DescribeStream",
+      "kinesis:ListStreams",
+    ]
+
+    resources = [
+      "${var.destination_stream_arn}",
+    ]
+  }
+
+  statement {
+    actions = [
+      "iam:PassRole",
+    ]
+
+    resources = [
+      "${aws_iam_role.flow_log_subscription_role.arn}",
+    ]
+  }
+}
+
+// IAM Policy Doc: Allow Cross Account Flow Logs
+data "aws_iam_policy_document" "cross_account_subscription_filter" {
+  count = "${length(var.cross_account_ids) > 0 ? 1 : 0}"
+
+  statement {
+    effect = "Allow"
```

Comment:
Can we add a statement ID `SID` here which is very clear about what it does? E.g. `sid = "AllowCrossAccountLogSubscription" or something

Reply:
There's a comment on the resource, is that still needed?

Reply:
The `SID` is attached to the actual policy, which I think can be searched / show up in alerts, etc. I don't have a strong preference either way